src/anndata/tests/helpers.py

```diff
@@ -82,30 +98,77 @@ def gen_vstr_recarray(m, n, dtype=None):
     )
 
 
-def gen_typed_df(n, index=None):
-    # TODO: Think about allowing index to be passed for n
-    letters = np.fromiter(iter(ascii_letters), "U1")
-    if n > len(letters):
-        letters = letters[: n // 2]  # Make sure categories are repeated
-    return pd.DataFrame(
-        {
-            "cat": pd.Categorical(np.random.choice(letters, n)),
-            "cat_ordered": pd.Categorical(np.random.choice(letters, n), ordered=True),
-            "int64": np.random.randint(-50, 50, n),
-            "float64": np.random.random(n),
-            "uint8": np.random.randint(255, size=n, dtype="uint8"),
-            "bool": np.random.randint(0, 2, size=n, dtype=bool),
-            "nullable-bool": pd.arrays.BooleanArray(
+def issubdtype(
+    a: np.dtype | pd.api.extensions.ExtensionDtype | type, b: type[DT]
+) -> TypeGuard[DT]:
+    if isinstance(a, type) and issubclass(a, pd.api.extensions.ExtensionDtype):
+        return issubclass(a, b)
+    if isinstance(a, pd.api.extensions.ExtensionDtype):
+        return isinstance(a, b)
+    try:
+        return np.issubdtype(a, b)
+    except TypeError:
+        pytest.fail(f"issubdtype can’t handle everything yet: {a} {b}")
+
+
+def gen_random_column(
```

Comment:
Can we change the order of this function so that it goes

1. check the non-`n_bits` dtypes first
2. get the bits without an `AttributeError` (since presumably that exception was handled above in 1.)
3. Check the rest of the dtypes

Reply:
good idea!